front/src/components/TodoListItem/__tests__/TodoListItem.test.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { ITodoListItem } from '../TodoListItem';
+import TodoListItem from '../index';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { iconSymbols } from '../../../constants';
+import { Todo } from '../../../types';
+
+
+const defaultProps: ITodoListItem = {
+  todo: {
+    id: '1111',
+    title: 'To buy a pizza',
+    importance: 'ordinary',
+    isDone: false,
+  },
+  onClick: () => {},
+  onEdit: () => {},
+  onDelete: () => {},
+}
+
+const renderTodoListItem = (props: Partial<ITodoListItem> = {}) => {
+  const componentProps = { ...defaultProps, ...props };
+
+  return render(<TodoListItem {...componentProps} />);
```

Comment:
What about dropping extra variable `componentProps` and doing everything inline?

```
return render(<TodoListItem  {...defaultProps} {...props} />);
```
Imo it's not just shorter, but more clear "apply defaults, overrides with what we have just passed in". What do you think?

Reply:
@skyboyer 
I agree with you. I'm ready to use this style in my future code.
If you want to hear my opinion here: the reason why I wrote it in such style is that in my case we spread only a single object in the <TodoListItem>. Imo object spreading in a component doesn't look very good. Object spreading looks natural only within another object. But I think it will be better just to follow the common patterns.

Reply:
@skyboyer 
I'm just curious why people never do this like `return render(<TodoListItem  {...defaultProps, ...props} />)` ?

Reply:
regarding object spreading for component: it's a typical pattern but it somehow harms readability: 1) we need to switch onto variable(s) declarations to see which props are listed there and with which values; 2) especially, if there are 2+ objects are spread, we cannot be sure about same-name-property overriding each other depending on ordering

regarding

```
return render(<TodoListItem {...defaultProps, ...props} />)
```
vs
```
return render(<TodoListItem {...defaultProps} {...props} />)
```
2nd option is more flexible since we can add specific single props between them:

```
return render(<TodoListItem {...defaultProps} a="1" {...props} />)
```
which is not really clean, so I'd not suggest to use that. But also we can move prop on separate lines more easily:

```
return render(<TodoListItem 
  {...defaultProps}
  {...props} 
/>)
```
Compare this to:
```
return render(<TodoListItem 
  {
    ...defaultProps, 
    ...props
  } 
/>)
```
Or probably, I get used to see first option and it's my bias 🤷